`e2e` nightly fail sometimes because some injected evidence is lost

## Bug Report

### Setup

**CometBFT version** (use `cometbft version` or `git rev-parse --verify HEAD` if installed from source): `v1.x`

**Have you tried the latest version**: yes

**ABCI app** (name for built-in, URL for self-written if it's publicly available):
e2e app

**Environment**:
N/A

**node command runtime flags**:
N/A

### Config

According to nightly `e2e` manifest: networks/nightly/gen-group01-0004.toml

### What happened?

Some evidences are lost during nightly execution:

```
--- FAIL: TestEvidence_Misbehavior (0.00s)
    evidence_test.go:20: 
        	Error Trace:	/home/runner/work/cometbft/cometbft/test/e2e/tests/evidence_test.go:20
        	Error:      	Not equal: 
        	            	expected: 10
        	            	actual  : 4
        	Test:       	TestEvidence_Misbehavior
        	Messages:   	difference between the amount of evidence produced and committed
FAIL
FAIL	github.com/cometbft/cometbft/test/e2e/tests	23.159s
FAIL
```

### What did you expect to happen?

No failure, all evidences accepted

### How to reproduce it

See first commit of PR fixing the problem

### Logs

See above

### `dump_consensus_state` output

N/A

### Anything else we need to know

Root cause identified, fix exists, PR coming up

--- a/.github/workflows/e2e-manual-multiversion.yml
+++ b/.github/workflows/e2e-manual-multiversion.yml
@@ -34,12 +34,12 @@ jobs:
         # based on the current branch as compared to the latest version.
         run: ./build/generator -g 5 -m "latest:1,local:2" -d networks/nightly/ -p
 
-      - name: Run ${{ matrix.p2p }} p2p testnets
+      - name: Run p2p testnets (${{ matrix.group }})
         if: matrix.group != 5
         working-directory: test/e2e
         run: ./run-multiple.sh networks/nightly/*-group${{ matrix.group }}-*.toml
 
-      - name: Run ${{ matrix.p2p }} p2p regression testnets
+      - name: Run p2p testnets (regression)
         if: matrix.group == 5
         working-directory: test/e2e
         run: ./run-multiple.sh networks_regressions/*.toml

--- a/.github/workflows/e2e-manual-multiversion.yml
+++ b/.github/workflows/e2e-manual-multiversion.yml
@@ -11,7 +11,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        group: ['00', '01', '02', '03', '04']
+        group: ['00', '01', '02', '03', '04', '05']
     runs-on: ubuntu-latest
     timeout-minutes: 60
     steps:
@@ -27,6 +27,7 @@ jobs:
         run: make -j2 docker generator runner tests
 
       - name: Generate testnets
+        if: matrix.group != 5
         working-directory: test/e2e
         # When changing -g, also change the matrix groups above
         # Generate multi-version tests with double the quantity of E2E nodes
@@ -34,5 +35,11 @@ jobs:
         run: ./build/generator -g 5 -m "latest:1,local:2" -d networks/nightly/ -p
 
       - name: Run ${{ matrix.p2p }} p2p testnets
+        if: matrix.group != 5
         working-directory: test/e2e
         run: ./run-multiple.sh networks/nightly/*-group${{ matrix.group }}-*.toml
+
+      - name: Run ${{ matrix.p2p }} p2p regression testnets
+        if: matrix.group == 5
+        working-directory: test/e2e
+        run: ./run-multiple.sh networks_regressions/*.toml

--- a/.github/workflows/e2e-manual.yml
+++ b/.github/workflows/e2e-manual.yml
@@ -11,7 +11,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        group: ['00', '01', '02', '03', '04']
+        group: ['00', '01', '02', '03', '04', '05']
     runs-on: ubuntu-latest
     timeout-minutes: 60
     steps:
@@ -27,10 +27,17 @@ jobs:
         run: make -j2 docker generator runner tests
 
       - name: Generate testnets
+        if: matrix.group != 5
         working-directory: test/e2e
         # When changing -g, also change the matrix groups above
         run: ./build/generator -g 5 -d networks/nightly/ -p
 
-      - name: Run ${{ matrix.p2p }} p2p testnets
+      - name: Run p2p testnets (${{ matrix.group }})
+        if: matrix.group != 5
         working-directory: test/e2e
         run: ./run-multiple.sh networks/nightly/*-group${{ matrix.group }}-*.toml
+
+      - name: Run p2p testnets (regression)
+        if: matrix.group == 5
+        working-directory: test/e2e
+        run: ./run-multiple.sh networks_regressions/*.toml

--- a/.github/workflows/e2e-nightly-1x.yml
+++ b/.github/workflows/e2e-nightly-1x.yml
@@ -15,7 +15,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        group: ['00', '01', '02', '03', "04"]
+        group: ['00', '01', '02', '03', '04', '05']
     runs-on: ubuntu-latest
     timeout-minutes: 60
     steps:
@@ -38,13 +38,20 @@ jobs:
         run: make -j2 docker generator runner tests
 
       - name: Generate testnets
+        if: matrix.group != 5
         working-directory: test/e2e
         # When changing -g, also change the matrix groups above
         run: ./build/generator -g 5 -d networks/nightly/ -p
 
-      - name: Run ${{ matrix.p2p }} p2p testnets
+      - name: Run p2p testnets (${{ matrix.group }})
+        if: matrix.group != 5
         working-directory: test/e2e
         run: ./run-multiple.sh networks/nightly/*-group${{ matrix.group }}-*.toml
+
+      - name: Run p2p testnets (regression)
+        if: matrix.group == 5
+        working-directory: test/e2e
+        run: ./run-multiple.sh networks_regressions/*.toml
 
     outputs:
       git-branch: ${{ steps.git-info.outputs.branch }}

--- a/.github/workflows/e2e-nightly-38x.yml
+++ b/.github/workflows/e2e-nightly-38x.yml
@@ -15,7 +15,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        group: ['00', '01', '02', '03', "04"]
+        group: ['00', '01', '02', '03', '04', '05']
     runs-on: ubuntu-latest
     timeout-minutes: 60
     steps:
@@ -38,13 +38,20 @@ jobs:
         run: make -j2 docker generator runner tests
 
       - name: Generate testnets
+        if: matrix.group != 5
         working-directory: test/e2e
         # When changing -g, also change the matrix groups above
         run: ./build/generator -g 5 -d networks/nightly/ -p
 
-      - name: Run ${{ matrix.p2p }} p2p testnets
+      - name: Run p2p testnets (${{ matrix.group }})
+        if: matrix.group != 5
         working-directory: test/e2e
         run: ./run-multiple.sh networks/nightly/*-group${{ matrix.group }}-*.toml
+
+      - name: Run p2p testnets (regression)
+        if: matrix.group == 5
+        working-directory: test/e2e
+        run: ./run-multiple.sh networks_regressions/*.toml
 
     outputs:
       git-branch: ${{ steps.git-info.outputs.branch }}

--- a/.github/workflows/e2e-nightly-main.yml
+++ b/.github/workflows/e2e-nightly-main.yml
@@ -16,7 +16,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        group: ['00', '01', '02', '03', "04"]
+        group: ['00', '01', '02', '03', '04', '05']
     runs-on: ubuntu-latest
     timeout-minutes: 60
     steps:
@@ -32,13 +32,20 @@ jobs:
         run: make -j2 docker generator runner tests
 
       - name: Generate testnets
+        if: matrix.group != 5
         working-directory: test/e2e
         # When changing -g, also change the matrix groups above
         run: ./build/generator -g 5 -d networks/nightly/ -p
 
-      - name: Run ${{ matrix.p2p }} p2p testnets
+      - name: Run p2p testnets (${{ matrix.group }})
+        if: matrix.group != 5
         working-directory: test/e2e
         run: ./run-multiple.sh networks/nightly/*-group${{ matrix.group }}-*.toml
+
+      - name: Run p2p testnets (regression)
+        if: matrix.group == 5
+        working-directory: test/e2e
+        run: ./run-multiple.sh networks_regressions/*.toml
 
   e2e-nightly-fail:
     needs: e2e-nightly-test

--- a/test/e2e/generator/generate.go
+++ b/test/e2e/generator/generate.go
@@ -49,7 +49,7 @@ var (
 		4 * int(e2e.EvidenceAgeHeight),
 	}
 	nodeEnableCompanionPruning = uniformChoice{true, false}
-	evidence                   = uniformChoice{0, 1, 10}
+	evidence                   = uniformChoice{0, 1, 10, 20, 200}
 	abciDelays                 = uniformChoice{"none", "small", "large"}
 	nodePerturbations          = probSetChoice{
 		"disconnect": 0.1,

--- a/test/e2e/networks/ci.toml
+++ b/test/e2e/networks/ci.toml
@@ -1,102 +1,16 @@
-# This testnet is run by CI, and attempts to cover a broad range of
-# functionality with a single network.
-
-ipv6 = true
-initial_height = 1000
-vote_extensions_update_height = 1004
-vote_extensions_enable_height = 1007
-pbts_update_height = 1006
-pbts_enable_height = 1009
-evidence = 5
-initial_state = { initial01 = "a", initial02 = "b", initial03 = "c" }
-prepare_proposal_delay = "100ms"
-process_proposal_delay = "100ms"
-check_tx_delay = "0ms"
-# The most common case (e.g. Cosmos SDK-based chains).
-abci_protocol = "builtin"
+evidence = 120
 prometheus = true
-peer_gossip_intraloop_sleep_duration = "50ms"
-abci_tests_enabled = true
+pbts_enable_height = 1
 
 [validators]
-validator01 = 100
+  validator01 = 33
+  validator02 = 67
 
-[validator_update.0]
-validator01 = 10
-validator02 = 20
-validator03 = 30
-validator04 = 40
+[node]
+  [node.validator01]
+    mode = "validator"
+    persistent_peers = ["validator02"]
+    clock_skew = "40s"
+  [node.validator02]
+    mode = "validator"
 
-[validator_update.1010]
-validator05 = 50
-
-# validator03 gets killed and validator05 has lots of perturbations, so weight them low.
-[validator_update.1020]
-validator01 = 100
-validator02 = 100
-validator03 = 50
-validator04 = 100
-validator05 = 50
-
-[node.seed01]
-mode = "seed"
-perturb = ["restart"]
-
-[node.validator01]
-seeds = ["seed01"]
-snapshot_interval = 5
-perturb = ["disconnect"]
-clock_skew = "20s"
-
-[node.validator02]
-seeds = ["seed01"]
-database = "pebbledb"
-privval_protocol = "tcp"
-persist_interval = 0
-perturb = ["restart"]
-
-[node.validator03]
-seeds = ["seed01"]
-database = "badgerdb"
-privval_protocol = "unix"
-persist_interval = 3
-retain_blocks = 10
-enable_companion_pruning = true
-perturb = ["kill"]
-
-[node.validator04]
-persistent_peers = ["validator01"]
-database = "rocksdb"
-perturb = ["pause"]
-
-[node.validator05]
-start_at = 1005 # Becomes part of the validator set at 1010
-persistent_peers = ["validator01", "full01"]
-database = "rocksdb"
-privval_protocol = "tcp"
-perturb = ["kill", "pause", "disconnect", "restart"]
-
-[node.full01]
-start_at = 1010
-mode = "full"
-persistent_peers = ["validator01", "validator02", "validator03", "validator04", "validator05"]
-retain_blocks = 10
-enable_companion_pruning = true
-perturb = ["restart"]
-
-[node.full02]
-start_at = 1015
-mode = "full"
-state_sync = true
-seeds = ["seed01"]
-perturb = ["restart"]
-
-[node.light01]
-mode= "light"
-start_at= 1005
-persistent_peers = ["validator01", "validator02", "validator03"]
-
-[node.light02]
-mode= "light"
-start_at= 1015
-persistent_peers = ["validator04", "full01", "validator05"]

--- a/test/e2e/networks/ci.toml
+++ b/test/e2e/networks/ci.toml
@@ -60,7 +60,7 @@ seeds = ["seed01"]
 database = "badgerdb"
 privval_protocol = "unix"
 persist_interval = 3
-retain_blocks = 10
+retain_blocks = 20
 enable_companion_pruning = true
 perturb = ["kill"]
 
@@ -80,7 +80,7 @@ perturb = ["kill", "pause", "disconnect", "restart"]
 start_at = 1010
 mode = "full"
 persistent_peers = ["validator01", "validator02", "validator03", "validator04", "validator05"]
-retain_blocks = 10
+retain_blocks = 20
 enable_companion_pruning = true
 perturb = ["restart"]
 

--- a/test/e2e/networks/ci.toml
+++ b/test/e2e/networks/ci.toml
@@ -1,16 +1,102 @@
-evidence = 120
+# This testnet is run by CI, and attempts to cover a broad range of
+# functionality with a single network.
+
+ipv6 = true
+initial_height = 1000
+vote_extensions_update_height = 1004
+vote_extensions_enable_height = 1007
+pbts_update_height = 1006
+pbts_enable_height = 1009
+evidence = 5
+initial_state = { initial01 = "a", initial02 = "b", initial03 = "c" }
+prepare_proposal_delay = "100ms"
+process_proposal_delay = "100ms"
+check_tx_delay = "0ms"
+# The most common case (e.g. Cosmos SDK-based chains).
+abci_protocol = "builtin"
 prometheus = true
-pbts_enable_height = 1
+peer_gossip_intraloop_sleep_duration = "50ms"
+abci_tests_enabled = true
 
 [validators]
-  validator01 = 33
-  validator02 = 67
+validator01 = 100
 
-[node]
-  [node.validator01]
-    mode = "validator"
-    persistent_peers = ["validator02"]
-    clock_skew = "40s"
-  [node.validator02]
-    mode = "validator"
+[validator_update.0]
+validator01 = 10
+validator02 = 20
+validator03 = 30
+validator04 = 40
 
+[validator_update.1010]
+validator05 = 50
+
+# validator03 gets killed and validator05 has lots of perturbations, so weight them low.
+[validator_update.1020]
+validator01 = 100
+validator02 = 100
+validator03 = 50
+validator04 = 100
+validator05 = 50
+
+[node.seed01]
+mode = "seed"
+perturb = ["restart"]
+
+[node.validator01]
+seeds = ["seed01"]
+snapshot_interval = 5
+perturb = ["disconnect"]
+clock_skew = "20s"
+
+[node.validator02]
+seeds = ["seed01"]
+database = "pebbledb"
+privval_protocol = "tcp"
+persist_interval = 0
+perturb = ["restart"]
+
+[node.validator03]
+seeds = ["seed01"]
+database = "badgerdb"
+privval_protocol = "unix"
+persist_interval = 3
+retain_blocks = 10
+enable_companion_pruning = true
+perturb = ["kill"]
+
+[node.validator04]
+persistent_peers = ["validator01"]
+database = "rocksdb"
+perturb = ["pause"]
+
+[node.validator05]
+start_at = 1005 # Becomes part of the validator set at 1010
+persistent_peers = ["validator01", "full01"]
+database = "rocksdb"
+privval_protocol = "tcp"
+perturb = ["kill", "pause", "disconnect", "restart"]
+
+[node.full01]
+start_at = 1010
+mode = "full"
+persistent_peers = ["validator01", "validator02", "validator03", "validator04", "validator05"]
+retain_blocks = 10
+enable_companion_pruning = true
+perturb = ["restart"]
+
+[node.full02]
+start_at = 1015
+mode = "full"
+state_sync = true
+seeds = ["seed01"]
+perturb = ["restart"]
+
+[node.light01]
+mode= "light"
+start_at= 1005
+persistent_peers = ["validator01", "validator02", "validator03"]
+
+[node.light02]
+mode= "light"
+start_at= 1015
+persistent_peers = ["validator04", "full01", "validator05"]

--- a/test/e2e/networks_regressions/evidence_fail.toml
+++ b/test/e2e/networks_regressions/evidence_fail.toml
@@ -1,0 +1,16 @@
+evidence = 120
+prometheus = true
+pbts_enable_height = 1
+
+[validators]
+  validator01 = 33
+  validator02 = 67
+
+[node]
+  [node.validator01]
+    mode = "validator"
+    persistent_peers = ["validator02"]
+    clock_skew = "40s"
+  [node.validator02]
+    mode = "validator"
+

--- a/test/e2e/pkg/testnet.go
+++ b/test/e2e/pkg/testnet.go
@@ -66,8 +66,8 @@ const (
 	PerturbationRestart    Perturbation = "restart"
 	PerturbationUpgrade    Perturbation = "upgrade"
 
-	EvidenceAgeHeight int64         = 7
-	EvidenceAgeTime   time.Duration = 500 * time.Millisecond
+	EvidenceAgeHeight int64         = 14
+	EvidenceAgeTime   time.Duration = 1500 * time.Millisecond
 )
 
 // Testnet represents a single testnet.

--- a/test/e2e/runner/evidence.go
+++ b/test/e2e/runner/evidence.go
@@ -123,6 +123,7 @@ func InjectEvidence(ctx context.Context, r *rand.Rand, testnet *e2e.Testnet, amo
 			}
 			return err
 		}
+		time.Sleep(5 * time.Second / time.Duration(amount))
 	}
 
 	// wait for the node to reach the height above the forged height so that

